lib/elixir_tic_tac_toe.ex

```diff
@@ -1,52 +1,36 @@
-import GetUserMove
-import BestComputerSpot
-import WinningCombo
-import GetTokens
+import GameLoop
 
 defmodule ElixirTicTacToe do
 
   def play_TTT(board) do
 
-    _game_loop(board, return_token())
+    game_token = _player_setup()
 
-  end
+    GameLoop._game_loop(board, game_token)
 
-  def _print_board(board) do
-    printed_board = "\n #{board[:spotOne]} | #{board[:spotTwo]} | #{board[:spotThree]} "<>
-                    "\n---+---+---\n "<>
-                    "#{board[:spotFour]} | #{board[:spotFive]} | #{board[:spotSix]} "<>
-                    "\n---+---+---\n "<>
-                    "#{board[:spotSeven]} | #{board[:spotEight]} | #{board[:spotNine]} \n"
+    _replay_game(game_token)
 
-    IO.puts printed_board
   end
 
-  def _game_loop(board, player_token) do
-
-    if is_game_over(board, player_token) == false do
-      IO.puts("\nYour Turn!")
-      _print_board(board)
-
-      user_board = _user_turn(board, player_token)
-
-      if is_game_over(user_board, player_token) == false do
-        IO.puts("Computer's Turn!")
-        _game_loop(_computer_turn(user_board, player_token), player_token)
-      end
+  def _player_setup() do
+    if GameLoop._yes_no_checker("Would you like to play as a new player? (Y/N): ") == "Y" do
+      GetTokens.return_new_player_token()
+    else
+      GetTokens.return_player_token()
     end
   end
 
-  def _computer_turn(board, player_token) do
-    computer_board = get_best_spot(board, player_token)
-    _print_board(computer_board)
-
-    computer_board
-  end
-
-  def _user_turn(board, player_token) do
-    user_board = get_move(board, player_token)
-    _print_board(user_board)
-
-    user_board
+  def _replay_game(token) do
+    if GameLoop._yes_no_checker("Do you want to play again? (Y/N): ") == "Y" do
+      clean_board = [
```

Comment:
I see the empty board defined here and then also inside Main. Would be nice for it to only be defined once. One way would be to define a function in this module that returns a fresh board. Then both Modules could call that function.

Reply:
return_clean_board has been created and injected in place of the manual declaration